src/ome.ts

```diff
@@ -0,0 +1,226 @@
+    const loader = loaders.find(Boolean);
+
+    // Load Image to use for channel names, rendering settings, sizeZ, sizeT etc.
+    const sourceData: SourceData = loadOME(config, imageAttrs.omero, loader as ZarrLoader);
+    const cols = Math.ceil(Math.sqrt(imagePaths.length));
+
+    if (acquisitions.length > 0) {
+        // To show acquisition chooser in UI
+        sourceData.acquisitions = acquisitions;
+        sourceData.acquisition = acquisition || '-1';
+    }
+
+    sourceData.loaders = loaders;
+    sourceData.name = `Well ${row}${col}`;
+    sourceData.rows = Math.ceil(imagePaths.length / cols);
+    sourceData.columns = cols;
+    sourceData.onClick = (info: any) => {
+        let gridCoord = info.gridCoord;
+        if (!gridCoord) {
+            return;
+        }
+        const { row, column } = gridCoord;
+        let { source } = sourceData;
+        if (typeof source === 'string' && !isNaN(row) && !isNaN(column)) {
+            const field = (row * cols) + column;
+            if (source.endsWith('/')) {
+                source = source.slice(0, -1);
+            }
+            let imgSource = `${source}/${imagePaths[field]}`;
+            window.open(window.location.origin + '?source=' + imgSource);
+        }
+    }
+
+    return sourceData;
+}
+
+export async function loadOMEPlate(
+        config: ImageLayerConfig,
+        store: HTTPStore,
+        rootAttrs: RootAttrs,
+        acquisition: string | undefined
+    ): Promise<SourceData> {
+    const plateAttrs = rootAttrs.plate as OmePlateData;
+    if (!('columns' in plateAttrs) || !('rows' in plateAttrs)) {
+        throw Error(`Plate .zattrs missing columns or rows`);
+    }
+
+    let rows: number = plateAttrs.rows.length;
+    let columns: number = plateAttrs.columns.length;
+    let rowNames: string[] = plateAttrs.rows.map(row => row.name);
+    let columnNames: string[] = plateAttrs.columns.map(col => col.name);
+    let wellPaths = plateAttrs.wells.map(well => well.path + (well.path.endsWith('/') ? '' : '/'));
+
+    // Fields are by index and we assume at least 1 per Well
+    let field = '0';
+
+    // TEMP: try to support plates with plate/acquisition/row/column hierarchy
+    // Where plate.acquisitions = [{'path': '0'}]
+    let acquisitionPath = plateAttrs?.acquisitions?.[0]?.path;
+    acquisitionPath = acquisitionPath ? acquisitionPath + '/' : '';
+
+    // imagePaths covers whole plate (not sparse) - but some will be '' if no Well
+    const imagePaths = rowNames.flatMap(row => {
+        return columnNames.map(col => {
+            let wellPath = `${acquisitionPath}${row}/${col}/`;
+            return wellPaths.includes(wellPath) ? `${wellPath}${field}/` : '';
+        });
+    });
+
+    // Find first valid Image, loading each Well in turn...
+    let imageAttrs = undefined;
+    async function getImageAttrs(path: string): Promise<any> {
+        if (path === '') return
+        try {
+            return await getJson(store, `${path}.zattrs`);
```

Comment:
hmm, I guess the real util we want is some `join` method:

```javascript
await getJson(store, join(path, '.zattrs'));
```

That would remove the need to ensure individual strings have trailing (or leading) `/` and just ensure the correct path when making requests.

Reply:
What do you think? `ensureEndsWith` or `join`? Either way, having a consistent way of dealing with this would be ideal.

Reply:
I've added a `join()` and `rstrip()` - belt and braces! But both are useful.